charts/k8s-monitoring/values.yaml

```diff
@@ -214,6 +214,68 @@ alloy-metrics:
   # @section -- Collectors - Alloy Metrics
   extraConfig: ""
 
+  # Remote configuration from a remote config server.
+  remoteConfig:
+    # -- Enable fetching configuration from a remote config server.
+    # @section -- Collectors - Alloy Metrics
+    enabled: false
+
+    # -- The URL of the remote config server.
+    # @section -- Collectors - Alloy Metrics
+    url: ""
+
+    auth:
+      # -- The type of authentication to use for the remote config server.
+      # @section -- Collectors - Alloy Metrics
+      type: "none"
+
+      # -- The username to use for the remote config server.
+      # @section -- Collectors - Alloy Metrics
+      username: ""
+      # -- The key for storing the username in the secret.
+      # @section -- Collectors - Alloy Metrics
+      usernameKey: "username"
+      # -- Raw config for accessing the password.
+      # @section -- Collectors - Alloy Metrics
+      usernameFrom: ""
```

Comment:
Do these get used somewhere?

Reply:
Yeah, because this line:
```
username = {{ include "secrets.read" (dict "object" . "key" "auth.username" "nonsensitive" true) }}
```
Actually does a ton.

If your values are:
```
auth: { username = "myuser" }
```
Then, this will create a K8s secret, with `username: myuser` and then the alloy config references that secret.

If your values are:
```
auth: { username = "myuser" }
secret: { embed = true }
```
Then the string "myuser" is set in the alloy config and no secret is created.

Finally, if your values are:
```
auth: { usernameFrom = env("MY_USERAME") }
```
Then it injects the raw string env("MY_USERAME") as the value for the username.

All of this is handled by the template functions.

Reply:
That's not even mentioning these values:
```
auth: { usernameKey = "user" }
secret: { create = false, name = mysecret }
```
Which then references another pre-created secret with a custom key in that secret.